storage/google_cloud/google_cloud_utils.py

```diff
@@ -88,5 +90,28 @@ def upload_file_to_blob(bucket_credentials_file_path, target_blob_url, f):
     log.info(f"Uploading file to blob '{target_blob_url}'...")
     storage_client = storage.Client.from_service_account_json(bucket_credentials_file_path)
     blob = _blob_at_url(storage_client, target_blob_url)
-    blob.upload_from_file(f)
-    log.info(f"Uploaded file to blob")
+
+    try:
+        blob.upload_from_file(f)
+        log.info(f"Uploaded file to blob")
+
+    except HttpError or socket.timeout as ex:
+        if ex.resp.status not in [408, 504]:
+            raise ex
+
+        log.warning(f"Failed to upload due to {ex.resp.status}")
+        chunk_sizes = [50, 25, 12.5, 6.25]
```

Comment:
Calling these the chunk_sizes is a bit confusing given that they're going to be multiplied by 1024^2. Adding a unit of MiB to the variable name would help.